### PR TITLE
[MRG] Switch to use TRAVIS_COMMIT instead of TRAVIS_PULL_REQUEST_SHA

### LINKED ISSUE #1777: Docker CI build broken
The latest merge seems to have broken @mr-c's latest contributions for Docker builds in Travis (#1333). See https://travis-ci.org/dib-lab/khmer/jobs/272705892.

### PR DESCRIPTION
**Pull request by betatim**
Fixes #1777 by switching from TRAVIS_PULL_REQUEST_SHA which can be empty to TRAVIS_COMMIT.

This might also fix the build error in #1801.

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?

### CODE REVIEW

**Comment by betatim** [comment]
Ready for review! @luizirber, @camillescott, @standage , or @ctb

WDYT?

**Comment by betatim** [comment]
Who knows, but given this is software merging this might cause robots to gain consciousness and boot Skynet?

**Comment by betatim** [comment]
Does someone remember why there is https://github.com/dib-lab/khmer/blob/dad0945c9671479a4077efd4ccc12332e1ccaa61/docker/Dockerfile#L20 with explicit include paths in the `Dockerfile` instead of using `make`?

**Comment by standage** [comment]
Might be a question for @mr-c, who implemented this IIRC.

**Comment by mr-c** [comment]
@betatim that version uses the zlib and libbz2 from the base OS image instead of the bundled code

**Comment by betatim** [comment]
Ah Ok. I added a note to `setup.cfg` to send people to the `Dockerfile` to keep things in sync.

File: .travis.yml [hunk 1]
```diff
@@ -87,7 +87,9 @@ script:
   - make py-demos
   - make cpp-demos
   - cd examples/stamps && ./do.sh
-  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then cd ../.. && COMMIT=$TRAVIS_PULL_REQUEST_SHA make docker-container; fi
+  # ${TRAVIS_PULL_REQUEST_SHA:-$TRAVIS_COMMIT} makes sure we get the commit
+  # that travis is testing during a PR and when we build master
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then cd ../.. && COMMIT=${TRAVIS_PULL_REQUEST_SHA:-$TRAVIS_COMMIT} make docker-container; fi
```
Comment (standage) [inline .travis.yml]: Ooooooh. This explains why all of the PR builds were passing but none of the merged builds were.

**Pull request merged**